DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc

```diff
@@ -573,35 +577,39 @@ void HcalRecHitsAnalyzer::analyze(edm::Event const& ev, edm::EventSetup const& c
     Handle<EBRecHitCollection> rhitEB;
 
 
-      ev.getByToken(tok_EB_, rhitEB);
-
-    EcalRecHitCollection::const_iterator RecHit = rhitEB.product()->begin();  
-    EcalRecHitCollection::const_iterator RecHitEnd = rhitEB.product()->end();  
-    
-    for (; RecHit != RecHitEnd ; ++RecHit) {
-       
-      double en  = RecHit->energy();
-      eEcal  += en;
-      eEcalB += en;
+    EcalRecHitCollection::const_iterator RecHit;
+    EcalRecHitCollection::const_iterator RecHitEnd;
 
+    if ( ev.getByToken(tok_EB_, rhitEB) ) {
 
+      RecHit = rhitEB.product()->begin();  
```

Comment:
this can all be reduced to 

for ( const auto & recHit : rhitEB.product() ) {
 double en = recHit->energy();...